src/envs/__init__.py

```diff
@@ -189,3 +189,27 @@ def get_stats(self):
 
 
 REGISTRY["gymma"] = partial(env_fn, env=_GymmaWrapper)
+
+
```

Comment:
I think we should make this more configurable in the GymmaWrapper. As you implement it, one can either specify to use the previous wrapper (with `gymma`) which will run everything with a single common reward, or you can use `gymmai` which will run everything with the default rewards but it doesn't ensure that rewards are now 1 per agent since it depends on the environment. For SMAC, `gymmai` will give only one reward, for LBF one per agent etc.

I think for that reason, this isn't a great unification. Two other things that are currently unsatisfying to me are
1. when you run experiments using e.g. `qmix` or `vdn` (which assume common rewards) you can still specify `gymmai` or `gymma` but in the former it will likely crash with some weird error. I think we should have an assert somewhere that verifies that `gymmai` is only used with algorithms that support individual rewards and give informative errors.
2. I don't see any changes on the algorithm side. Are you very sure that PPO/ A2C work as expected in the case of individual rewards with `gymmai`? This might be the case as a result of broadcasting, but it is a slippery slope and it's easy to have hard-to-spot errors since we previously assumed there to be only a scalar reward. I'd definitely want to carefully go through that and see what happens in all cases.

Reply:
In my opinion, the ideal way of implementing this would be to have a `common_reward` and a `reward_scalarisation` config value. The former is a boolean and specifies whether common rewards should be used or not, and the latter specifies how to scalarise multiple rewards into one.

We could give these as arguments to the `gymma` wrapper and it combines rewards or doesn't as required to match these specifications. We would also need to make sure that we handle SMAC correctly which doesn't use the gymma wrapper at all (could assert that `common_reward == true` since SMAC only supports that).

Additionally, now we could have a simple list of which algorithms support non common reward setting and assert that an algorithm is compatible with the type of reward function we are trying to use.

This would all hopefully make it more robust, extendable without introducing much copy-paste code.

Reply:
@LukasSchaefer Thanks for the feedback. I wouldn't mind changing the `gymmai` config to the one above as long as it has backward compatibility. At the moment I have decided to seprate these changes in seprate branches (one for independant rewards and the other for critic RNN feature) to simplify the changes. Later, we can add the two changes as seprate PRs. These seprate branches can be found in the fork on my accound with names `indRewardsOnly` and `rnncritic`.